### PR TITLE
Refactoring poll code

### PR DESCRIPTION
**Pull request by pseudomonas**
First stab at refactoring. The code hasn't been beautified, just moved out into different modules so that there's (hopefully) no if `($question->type eq "tickybox") { ...` left in `Poll.pm` or `Question.pm`

### CODE REVIEW

File: cgi-bin/LJ/Poll/Question.pm [hunk 1]
```diff
@@ -15,6 +15,26 @@ package LJ::Poll::Question;
...
+
+### I don't quite see why I should need to do the use here, but it does
+### seem to be necessary - asb
+use LJ::Poll::Question::CheckBox;
+use LJ::Poll::Question::RadioButton;
+use LJ::Poll::Question::MultiChoice;
+use LJ::Poll::Question::Scale;
+use LJ::Poll::Question::DropDown;
+use LJ::Poll::Question::Text;
+
```
Comment (kareila) [inline cgi-bin/LJ/Poll/Question.pm]: This is common and annoying enough that we have a shortcut:

`LJ::ModuleLoader->require_subclasses( "LJ::Poll::Question" );`

File: etc/config.pl [hunk 2]
```diff
@@ -349,7 +349,7 @@
             google_analytics => 0,
             'inbox_max' => 2000,
             'interests' => 150,
-            'makepoll' => 0,
+            'makepoll' => 1,
```
Comment (kareila) [inline etc/config.pl]: I'm sure you needed to change this for your testing, but it probably shouldn't be included in the pull request.

Comment (rahaeli) [inline etc/config.pl]: Always a good idea to put etc/config*.pl in your .gitignore! It means manually updating the configs if we add a new config variable, but it prevents having to be paranoid about what changes creep in.

**Comment by kareila** [comment]
I'm really excited that you did this! I left a few style comments based on my initial impressions, and will take a closer look at the functional changes on my next pass.

File: etc/config.pl (hunk 2 above)
Comment (pseudomonas) [inline etc/config.pl]: aha, the gitignore was something I didn't know about. Good plan!

**Comment by pseudomonas** [comment]
@kareila thanks so much for your comments! The refactoring has actually made it really quite easy to make a new poll type by creating a new subclass of `LJ::Poll::Question::MultiChoice` so I think it's served its purpose for me at least.

**Comment by pseudomonas** [comment]
Oh, bottoms. The push of the latest code (implementing runoff polls) should probably be separated from the refactoring per se.

**Comment by pseudomonas** [comment]
I'm still getting used to git :disappointed:

**Comment by zorkian** [comment]
(Form letter incoming.)

Last update was a while ago and there are outstanding requested changes -- we'd love to see this functionality, but it will probably need some TLC to get up to scratch. If someone does want to pick this back up we'd love to chat! In the meantime, I'm closing very old PRs to keep the board more clean.
